api/src/main/java/org/pmiops/workbench/access/AccessSyncServiceImpl.java

```diff
@@ -60,11 +71,36 @@ public DbUser updateUserAccessTiers(DbUser dbUser, Agent agent) {
     final List<DbAccessTier> previousAccessTiers = accessTierService.getAccessTiersForUser(dbUser);
 
     final List<DbAccessTier> newAccessTiers = getUserAccessTiersList(dbUser);
+
+    boolean enableInitialCreditsExpiration =
+        workbenchConfigProvider.get().featureFlags.enableInitialCreditsExpiration;
+    long freeTierCreditValidityPeriodDays =
+        workbenchConfigProvider.get().billing.freeTierCreditValidityPeriodDays;
+
     if (!newAccessTiers.equals(previousAccessTiers)) {
       userServiceAuditor.fireUpdateAccessTiersAction(
           dbUser, previousAccessTiers, newAccessTiers, agent);
     }
 
+    if (enableInitialCreditsExpiration) {
```

Comment:
this is quite surprising to see without additional context.  Could you add comments describing the purpose of this section as well as detailing why this particular logic (89-91) was chosen?